src/components/scenes/newAccount/NewAccountTosScene.tsx

```diff
@@ -210,3 +167,124 @@ const getStyles = cacheStyles((theme: Theme) => ({
     minHeight: theme.rem(6)
   }
 }))
+
+const setTouchOtp = async (account: EdgeAccount, dispatch: Dispatch) => {
+  await enableTouchId(account).catch(e => {
+    console.log(e) // Fail quietly
+  })
+  await account.dataStore.setItem(
+    Constants.OTP_REMINDER_STORE_NAME,
+    Constants.OTP_REMINDER_KEY_NAME_CREATED_AT,
+    Date.now().toString()
+  )
+  dispatch(loadTouchState())
+}
+
+/**
+ * Terms of Service scene for new regular or light accounts
+ */
+interface NewAccountTosProps extends SceneProps<'newAccountTos'> {
+  branding: Branding
+}
+export const NewAccountTosScene = (props: NewAccountTosProps) => {
+  const { route, branding } = props
+  const imports = useImports()
+  const { context } = imports
+  const dispatch = useDispatch()
+
+  const handleBack = useHandler((): void => {
+    dispatch({
+      type: 'NAVIGATE',
+      data: { name: 'newAccountPin', params: { ...route.params } }
+    })
+  })
+
+  const handleNext = useHandler(async () => {
+    logEvent(`Signup_Terms_Agree_and_Create_User`)
+    const { username, password, pin } = route.params
+
+    const account = await context.createAccount({
+      ...imports.accountOptions,
+      username,
+      password,
+      pin
+    })
+    account.watch('loggedIn', loggedIn => {
+      if (!loggedIn) dispatch({ type: 'RESET_APP' })
+    })
+    await setTouchOtp(account, dispatch)
+
+    dispatch({
+      type: 'NAVIGATE',
+      data: {
+        name: 'newAccountReview',
+        params: {
+          ...route.params,
+          account
+        }
+      }
+    })
+
+    logEvent('Signup_Create_User_Success')
+  })
+
+  return (
+    <TosComponent
+      hidePasswordTerms={route.params.username == null}
+      branding={branding}
+      onBack={handleBack}
+      onNext={handleNext}
+    />
+  )
+}
+
+/**
+ * Terms of Service scene for upgrading existing (light) accounts
+ */
+interface UpgradeTosProps extends SceneProps<'upgradeTos'> {
+  branding: Branding
+}
+export const UpgradeTosScene = (props: UpgradeTosProps) => {
+  const { route, branding } = props
+  const dispatch = useDispatch()
+
+  const handleBack = useHandler((): void => {
+    dispatch({
+      type: 'NAVIGATE',
+      data: { name: 'upgradeUsername', params: { ...route.params } }
+    })
+  })
+
+  const handleNext = useHandler(async () => {
+    logEvent(`Signup_Terms_Agree_and_Back_Up_User`)
```

Comment:
Maybe change "Signup" to "Upgrade".

Reply:
I'm trying to keep the naming understood by the users of the analytics (Peter) but I agree it's awkward. Honestly it should just be a param on a parent event.

Reply:
After speaking with Peter, we have settled on some tweaks to the tracking events and naming, will add as a separate commit